linkerd/metrics/src/new_metrics.rs

```diff
@@ -33,28 +33,29 @@ where
             .field("store", &self.store)
             .field("inner", &self.inner)
             .field("svc", &format_args!("PhantomData<{}>", type_name::<S>()))
+            .field("param", &format_args!("PhantomData<{}>", type_name::<S>()))
             .finish()
     }
 }
 
-impl<N, K, M, S, T> svc::NewService<T> for NewMetrics<N, K, M, S>
+impl<N, K, M, S, T, P> svc::NewService<T> for NewMetrics<N, K, M, S, P>
 where
-    T: svc::Param<K>,
+    T: svc::Param<P>,
     N: svc::NewService<T>,
     S: From<(N::Service, Arc<M>)>,
     M: Default,
-    K: Hash + Eq,
+    K: Hash + Eq + From<P>,
```

Comment:
More of a question than a suggestion: but what would it mean to avoid the `P` type? I.e. if we just required that `T: P: svc::Param<K>`, how would that change things?

This is a bit heady, so it would be nice to simplify this; but I don't have a great understanding of the tradeoff.

Reply:
I wonder if we can eliminate the TargetAddr type by always using `OrigDstAddr`? The TCP accept errors will _always_ be associated with the orig dst addr.

Reply:
@hawkw Some proposed simplifications here https://github.com/linkerd/linkerd2-proxy/pull/1147